### PR TITLE
Increase Gameroom paging limit

### PR DESCRIPTION
**Pull request by shannynalayna**
Increases the default paging limit to 1000 and also fixes the tests to reflect this modification.

### CODE REVIEW

File: examples/gameroom/daemon/src/rest_api/routes/mod.rs [hunk 1]
```diff
@@ -174,44 +174,46 @@ mod tests {
...
+            create_test_paging_response(DEFAULT_LIMIT, DEFAULT_OFFSET, 1000, 0, 9000);
         assert_eq!(test_paging_response, generated_paging_response);
     }
 
     #[test]
     fn test_50offset_paging_response() {
-        // Create paging response from default limit, offset of 50, and a total of 1000
-        let test_paging_response = get_response_paging_info(DEFAULT_LIMIT, 50, TEST_LINK, 1000);
-        let generated_paging_response = create_test_paging_response(DEFAULT_LIMIT, 50, 150, 0, 900);
+        // Create paging response from default limit, offset of 500, and a total of 10000
```
Comment (dplumb94) [inline examples/gameroom/daemon/src/rest_api/routes/mod.rs]: Is it necessary to update the tests besides `test_default_paging_response()`? If so, the names of the tests should be updated as well

File: examples/gameroom/gameroom-app/src/store/api.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ export async function gameroomPropose(
 }
 
 export async function listGamerooms(): Promise<Gameroom[]> {
-  const response = await gameroomAPI.get('/gamerooms');
+  const response = await gameroomAPI.get('/gamerooms?limit=1000');
```
Comment (peterschwarz) [inline examples/gameroom/gameroom-app/src/store/api.ts]: Does the server side have a max limit?

Comment (shannynalayna) [reply examples/gameroom/gameroom-app/src/store/api.ts]: No, the limit gets passed directly to the database query. I don't see any limit maximum documented from diesel.

Comment (peterschwarz) [reply examples/gameroom/gameroom-app/src/store/api.ts]: We should add a maximum on the server-side for this, as well.  Perhaps in a separate commit.

File: examples/gameroom/daemon/src/rest_api/routes/mod.rs [hunk 3]
```diff
@@ -166,6 +166,15 @@ pub fn get_response_paging_info(
     }
 }
 
+pub fn validate_limit(limit: usize) -> i64 {
+    let max = 1000 as usize;
```
Comment (peterschwarz) [inline examples/gameroom/daemon/src/rest_api/routes/mod.rs]: This could be a `const`

**Pull request merged**